frame/nfts/src/features/attributes.rs

```diff
@@ -34,90 +35,273 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		let mut collection_details =
 			Collection::<T, I>::get(&collection).ok_or(Error::<T, I>::UnknownCollection)?;
 
-		if let Some(check_owner) = &maybe_check_owner {
-			ensure!(check_owner == &collection_details.owner, Error::<T, I>::NoPermission);
-		}
+		ensure!(
+			Self::is_valid_namespace(
+				&origin,
+				&namespace,
+				&collection,
+				&collection_details.owner,
+				&maybe_item,
+			)?,
+			Error::<T, I>::NoPermission
+		);
 
 		let collection_config = Self::get_collection_config(&collection)?;
-		match maybe_item {
-			None => {
-				ensure!(
-					collection_config.is_setting_enabled(CollectionSetting::UnlockedAttributes),
-					Error::<T, I>::LockedCollectionAttributes
-				)
-			},
-			Some(item) => {
-				let maybe_is_locked = Self::get_item_config(&collection, &item)
-					.map(|c| c.has_disabled_setting(ItemSetting::UnlockedAttributes))?;
-				ensure!(!maybe_is_locked, Error::<T, I>::LockedItemAttributes);
+		// for the `CollectionOwner` namespace we need to check if the collection/item is not locked
+		match namespace {
+			AttributeNamespace::CollectionOwner => match maybe_item {
+				None => {
+					ensure!(
+						collection_config.is_setting_enabled(CollectionSetting::UnlockedAttributes),
+						Error::<T, I>::LockedCollectionAttributes
+					)
+				},
+				Some(item) => {
+					let maybe_is_locked = Self::get_item_config(&collection, &item)
+						.map(|c| c.has_disabled_setting(ItemSetting::UnlockedAttributes))?;
+					ensure!(!maybe_is_locked, Error::<T, I>::LockedItemAttributes);
+				},
 			},
-		};
+			_ => (),
+		}
```

Comment:
I'm slightly surprised that this works without a semicolon after the closing `match` brace. 
But i'm fairly new to rust. My expectation was that not adding a semicolon here would mean an early return.

Reply:
You need a semicolon when you use `match` as an expression: let smth = match {};